c7n/resources/sagemaker.py

```diff
@@ -66,14 +68,107 @@ class resource_type(object):
         service = 'sagemaker'
         enum_spec = ('list_training_jobs', 'TrainingJobSummaries', None)
         detail_spec = (
-            'describe_training_job', 'TrainingJobName',
-            'TrainingJobName', None)
+            'describe_training_job', 'TrainingJobName', 'TrainingJobName', None)
         id = 'TrainingJobArn'
         name = 'TrainingJobName'
         date = 'CreationTime'
         dimension = None
         filter_name = None
 
+    permissions = (
+        'sagemaker:ListTrainingJobs', 'sagemaker:DescribeTrainingJobs',
+        'sagemaker:ListTags')
+
+    def __init__(self, ctx, data):
+        super(SagemakerJob, self).__init__(ctx, data)
+        self.queries = QueryFilter.parse(
+            self.data.get('query', [
+                {'StatusEquals': 'InProgress'}]))
+
+    def resources(self, query=None):
+        q = self.query_filter()
+        if q is not None:
+            query = query or {}
+            for i in range(0, len(q)):
+                query[q[i]['Name']] = q[i]['Value']
+        return super(SagemakerJob, self).resources(query=query)
+
+    def query_filter(self):
+        result = []
+        names = set()
+        for q in self.queries:
+            query_filter = q.query()
```

Comment:
your method is already called query_filter.  Try not to also used a local variable named query_filter.  it's not technically a bug, just could lead to one.

Reply:
that's a good point. i did have some reservations on naming it that